packages/components/src/widgets/LambdaWidget/FunctionChart/index.tsx

```diff
@@ -63,14 +65,23 @@ export const FunctionChart: FC<FunctionChartProps> = ({
   const signatures = fn.signatures();
   const domain = signatures[0][0]?.domain;
 
-  const min = domain?.min ?? settings.functionChartSettings.start;
-  const max = domain?.max ?? settings.functionChartSettings.stop;
+  const min: number = domain?.min ?? settings.functionChartSettings.start;
+  const max: number = domain?.max ?? settings.functionChartSettings.stop;
 
-  const xScale = SqLinearScale.create({ min, max });
+  let xScale;
+  const isDateRange = domain?._value.type === "DateRange";
+  if (isDateRange) {
+    xScale = SqDateScale.create({ min, max });
+  } else {
+    xScale = SqLinearScale.create({ min, max });
+  }
```

Comment:
Maybe add `SqDomain.toScale()` method? Then: `const xScale = domain?.toScale() : SqLinearScale.create({ min: settings..., max: settings... })`.

Reply:
Good point. 

I also changed it so that if there's a domain, but settings too - the settings come first, but won't make the range go outside of the domain.

<img width="830" alt="image" src="https://github.com/quantified-uncertainty/squiggle/assets/377065/649e8c9c-5d8f-484f-9f61-fa94db5c321f">

<img width="638" alt="image" src="https://github.com/quantified-uncertainty/squiggle/assets/377065/ecc664db-50a5-4e80-8aff-0132de966fc7">

Reply:
> I also changed it so that if there's a domain, but settings too - the settings come first, but won't make the range go outside of the domain.

Are you sure it's a good idea? If we do this, it won't be possible to override the range for functions that are not wrapped in plots, which I think was useful. https://squiggle-website-git-datescale-quantified-uncertainty.vercel.app/playground?v=dev#code=eNqrVkpJTUsszSlxzk9JVbJSStOosFKI1jU0MNBRABKxmgq2ChVKtQDxpArs

(Also, in this example, `min` is not updated by settings min, because you check for `min` instead of `min !== undefined`)

Reply:
Oh, good point, I was misunderstanding this bit.